lib/javascript.d.ts

```diff
@@ -195,10 +200,10 @@ declare global {
 		type Object = StateObject | ChannelObject | DeviceObject | OtherObject;
 		type PartialObject = PartialStateObject | PartialChannelObject | PartialDeviceObject | PartialOtherObject;
 
-		type GetStateCallback = (err: string | null, state?: State) => void;
-		type SetStateCallback = (err: string | null, id?: string) => void;
+		type GetStateCallback<T = any> = (err: string | null, state?: State<T>) => void;
+		type SetStateCallback<T = any> = (err: string | null, id?: string) => void;
 
-		type StateChangeHandler = (obj: ChangedStateObject) => void;
+		type StateChangeHandler<T = any> = (obj: ChangedStateObject<T>) => void;
```

Comment:
This I don't like. You can achieve the same by just asserting the type on the callback parameter, e.g.:
```ts
on("state.id", (obj: iobJS.ChangedStateObject<boolean, string>) => { /* ... */ });
```
This way the generic is more directly associated with the parameter it belongs to. With a generic type on the method itself, it is not immediately clear that it influences a parameter in the callback method.